src/workerd/jsg/util.c++

```diff
@@ -90,9 +92,53 @@ kj::String typeName(const std::type_info& type) {
   return kj::mv(result);
 }
 
+namespace {
+
+// For internal errors, we generate an ID to include when rendering user-facing "internal error"
+// exceptions and internal logs, to make it easier to search for errors corresponding to user
+// reports.
+//
+// We'll use an ID of 24 base-32 encoded characters, just because its relatively simple to
+// generate from random bytes.  This should give us a value with 120 bits of uniqueness, which is
+// about as good as a UUID.
+//
+// (We're not using base-64 encoding to avoid issues with case insensitive search, as well as
+// ensuring that the id is easy to select and copy via double-clicking.)
+using InternalErrorId = kj::FixedArray<char, 24>;
+
+constexpr char BASE32_DIGITS[] = "0123456789abcdefghijklmnopqrstuv";
+
+InternalErrorId makeInternalErrorId() {
+  InternalErrorId id;
+  if (isPredictableModeForTest()) {
+    // In testing mode, use content that generates a "0123456789abcdefghijklm" ID:
+    for (auto i: kj::indices(id)) {
+      id[i] = i;
+    }
+  } else {
+    KJ_ASSERT(RAND_bytes(id.asPtr().asBytes().begin(), id.size()) == 1);
+  }
+  for (auto i: kj::indices(id)) {
+    id[i] = BASE32_DIGITS[static_cast<unsigned char>(id[i]) % 32];
+  }
+  return id;
+}
+
+kj::String renderInternalError(InternalErrorId& wdErrId) {
+  // TODO(now): put "internal error" change behind autogate or compatibility flag?
+  //
+  // It's possible that existing user error handling systems could rely on an exact match with the
+  // existing "internal error" string.  On the other hand, feature flags need a jsg::Lock to read,
+  // which may not be available in all contexts that generate internal errors?
+  return kj::str("internal error (wderrid ", wdErrId, ")");
```

Comment:
I'd suggest a format like:

`internal error; reference = 0123456789abcdefghijklmn`

Because:
* `wderrid` looks almost like a random string of letters, I don't think most people will figure out that it means "workerd error ID". "reference" feels like it has an intuitive meaning, though I'd be open to other suggestions.
* If somehow the "internal error" itself gets reported to Sentry, the format `; identifier = value` will be parsed as a parameter so that the value is not part of the fingerprint.

Reply:
Ah, OK... I wasn't sure if it was maybe preferable to include a relatively unique term, to make it easier to search for it in logs, or find information about online.  Sure, I can update the format; I'd expect user-facing errors like "internal error" to rarely go through the sentry handler, but it's possible there could be edge cases.

Reply:
I updated the PR to use the suggested format for the user-facing exception.  I'm preserving the "wderrid" name in internal logs to make it easier to search for, assuming there's not a preferable term.